// simplyemail/render.js
var fs = require('fs');
var jade = require('jade');
var inlineCss = require('inline-css');
var date = new Date();

module.exports = {
	cijfer: function(vak, vakUrl, settingsUrl, cijfer, gemiddelde, callback){
		return new Promise(function(resolve, reject){
			if(cijfer<5.5){
				var onv = true;
			}else{
				var onv = false;
			}
			var html = jade.renderFile(__dirname+"/"+"cijfer.jade", {
				vak:vak,
				vakUrl: vakUrl,
				settingsUrl: settingsUrl,
				cijfer: cijfer.toString().replace(".", ","),
				gemiddelde: gemiddelde.toString().replace(".", ","),
				onv: onv,
				year: date.getFullYear()
			});
			inlineCss(html,{
				preserveMediaQueries: true,
				url: "/"
			}).then(function(html){
				resolve(html);
			});
		});
	},
	project: function(project, projectUrl, settingsUrl, person, callback){
		return new Promise(function(resolve, reject){
			var html = jade.renderFile(__dirname+"/"+"project.jade", {
				project:project,
				projectUrl: projectUrl,
				settingsUrl: settingsUrl,
				person: person,
				year: date.getFullYear()
			});
			inlineCss(html,{
				preserveMediaQueries: true,
				url: "/"
			}).then(function(html){
				resolve(html);
			});
		});
	}
}
